AdobeStockClient/Model/Client.php

```diff
@@ -124,6 +119,39 @@ public function search(SearchCriteriaInterface $searchCriteria): SearchResultInt
         return $searchResult;
     }
 
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return \AdobeStock\Api\Response\SearchFiles
+     */
+    private function processSearchRequest(SearchCriteriaInterface $searchCriteria)
+    {
+        try {
+            $searchParams = $this->searchParametersProvider->apply($searchCriteria, new SearchParameters());
+
+            $resultsColumns = Constants::getResultColumns();
+            $resultColumnArray = [];
+            foreach ($this->config->getSearchResultFields() as $field) {
+                $resultColumnArray[] = $resultsColumns[$field];
+            }
+
+            $searchRequest = new SearchFilesRequest();
+            $searchRequest->setLocale($this->localeResolver->getLocale());
+            $searchRequest->setSearchParams($searchParams);
+            $searchRequest->setResultColumns($resultColumnArray);
+
+            $client = $this->getClient()->searchFilesInitialize($searchRequest, $this->getAccessToken());
+            $response = $client->getNextResponse();
+
+            return $response;
+        } catch (\Exception $exception) {
+            $this->exceptionManager->processException(
+                $exception,
+                ExceptionManagerInterface::FORBIDDEN_CONNECTION_ERROR_CODE
```

Comment:
Why are we trying to handle all possible problems in the code as a Forbidden error? In this case, we can just throw Localized exception without any managers

Reply:
I extended this part: we handle AdobeStock exception first, then Exception and if something extraordinary happened just throw a default exception.